base/bucket_gocb.go

```diff
@@ -1617,12 +1617,22 @@ func (bucket *CouchbaseBucketGoCB) Update(k string, exp uint32, callback sgbucke
 	}
 }
 
+func (bucket *CouchbaseBucketGoCB) WriteXattr(docKey string, xattrKey string, xattrVal interface{}) (uint64, error) {
```

Comment:
It looks like this is currently only being used by tests.  Is it eventually going to be used by non-test code, in scenarios not already covered by WriteCasWithXattr?

If it's test-only, I'd probably stick it in with test utilities instead of expanding the bucket API.  If it's non-test, it needs the additional handling that's present in places like WriteCasWithXattr.  (singleOps throttling, retry on recoverable write errors, etc).

Reply:
Moved into util_testing for GoCB testing